File: PixelDungeonPhotoMaker/src/au/net/genesis/mds/particles/FlyAway.java
package au.net.genesis.mds.particles;

import au.net.genesis.mds.helpers.MathHelper;

public class FlyAway extends ParticleLogic {
	
	private final int myRotation = MathHelper.randomRange(0, 360);
	private float speed = 0.5F;
	
	@Override
	public void initialise() {
		super.initialise();
		lifeTime = 25;
		alpha = 0;
		rotation = myRotation+90;
		xspeed = (float) MathHelper.xcomponent(speed, myRotation);
		yspeed = (float) MathHelper.ycomponent(speed, myRotation);
	}
	public void update() {
		super.update();
		xscale = (Math.abs((age % 10)-5)+1)/5F;
		if (age > lifeTime-10) {
			alpha = Math.max(alpha-0.1F, 0F);
		} else {
			alpha = Math.min(1F, alpha + 0.2F);
		}
	}

}

File: src/main/java/com/mdsimmo/particles/FlyAway.kt
package com.mdsimmo.particles

import com.mdsimmo.helpers.MathHelper
import kotlin.math.abs

class FlyAway : ParticleLogic() {

    override fun newParticle(): ParticleState {
        val rotation = MathHelper.randomRange(0, 360)
        val speed = 0.5f
        return ParticleState(
            lifeTime = 25,
            alpha = 0f,
            rotation = rotation + 90,
            xspeed = MathHelper.xcomponent(speed.toDouble(), rotation.toDouble()).toFloat(),
            yspeed = MathHelper.ycomponent(speed.toDouble(), rotation.toDouble()).toFloat()
        )
    }

    override fun update(particle: ParticleState): ParticleState {
        return super.update(particle).copy(
            xscale = (abs(particle.age % 10 - 5) + 1) / 5f,
            alpha = if (particle.age > particle.lifeTime - 10) {
                        (particle.alpha - 0.1f).coerceAtLeast(0f)
                    } else {
                        (particle.alpha + 0.2f).coerceAtMost(1f)
                    }
        )
    }
}